### PR TITLE
APEX-273 - Fix existing checkstyle violations in bufferserver module

### PR DESCRIPTION
**Pull request by vrozov**
@243826, @davidyan74, @tweise, @chandnisingh Please review

### CODE REVIEW

**Comment by pramodin** [comment]
Do we need to do the checkstyle separately now. Wasn't there a plan to do it as needed when folks are touching relavant code.

**Comment by vrozov** [comment]
This option lost in the community vote and I did not see any new vote, so I believe the plan is the same - take the pain and fix all code style violations.

**Comment by pramodin** [comment]
If the vote is to change it, can we use a special account like release to be the author of the changes so we can flag these as codestyle changes in future.

File: bufferserver/src/test/java/com/datatorrent/bufferserver/util/CodecTest.java [hunk 1]
```diff
@@ -22,8 +22,8 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testng.annotations.Test;
-import static org.testng.Assert.*;
 
+import static org.junit.Assert.assertEquals;
```
Comment (davidyan74) [inline bufferserver/src/test/java/com/datatorrent/bufferserver/util/CodecTest.java]: Not sure if we allow import static.  Should we just import org.junit.Assert and call Assert.assertEquals?

Comment (davidyan74) [inline bufferserver/src/test/java/com/datatorrent/bufferserver/util/CodecTest.java]: Also, should we get rid of all org.testing imports and use org.junit like the rest of the Apex core?

**Comment by vrozov** [comment]
Do we need a new special account? Will it be sufficient to use MalharJenkins?

**Comment by pramodin** [comment]
MalharJenkins will do

File: bufferserver/src/test/java/com/datatorrent/bufferserver/util/CodecTest.java (hunk 1 above)
Comment (vrozov) [inline bufferserver/src/test/java/com/datatorrent/bufferserver/util/CodecTest.java]: We do allow static import and use it in other classes (mostly in tests).

Bufferserver uses testng JUnit provider and it should be part of another exercise to migrate Bufferserver to org.junit from org.testng.

I will fix static import to org.testng.Assert.assertEquals.

**Comment by 243826** [comment]
when did the voting happen on this issue?

File: bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java [hunk 2]
```diff
@@ -139,13 +140,16 @@ public void rewind(final int baseSeconds, final int windowId) throws IOException
...
+     */
 
     final int numberOfInMemBlockPermits = this.numberOfInMemBlockPermits.addAndGet(numberOfInMemBlockRewound);
-    assert numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS : "Number of in memory block permits " + numberOfInMemBlockPermits + " exceeded configured maximum " + MAX_COUNT_OF_INMEM_BLOCKS + '.';
+    assert numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS : "Number of in memory block permits " +
+        numberOfInMemBlockPermits + " exceeded configured maximum " + MAX_COUNT_OF_INMEM_BLOCKS + '.';
     resumeSuspendedClients(numberOfInMemBlockPermits);
-    logger.debug("Discarded {} in memory blocks during rewind. Number of in memory blocks permits {} after rewinding {}. ", numberOfInMemBlockRewound, numberOfInMemBlockPermits, this);
+    logger.debug("Discarded {} in memory blocks during rewind. Number of in memory blocks permits {} after" +
+        " rewinding {}.", numberOfInMemBlockRewound, numberOfInMemBlockPermits, this);
```
Comment (tweise) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:152]: This change set illustrates in several cases how readability is affected by the line length enforcement. Breaking string literals to make checkstyle happy looks just wrong.

File: bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java [hunk 3]
```diff
@@ -177,11 +181,13 @@ public void reset()
...
   {
     final long longWindowId = (long)baseSeconds << 32 | windowId;
-    logger.debug("Purging {} from window ID {} to window ID {}", this, Codec.getStringWindowId(first.starting_window), Codec.getStringWindowId(longWindowId));
+    logger.debug("Purging {} from window ID {} to window ID {}", this, Codec.getStringWindowId(first.starting_window),
+        Codec.getStringWindowId(longWindowId));
 
     int numberOfInMemBlockPurged = 0;
     synchronized (this) {
-      for (Block prev = null, temp = first; temp != null && temp.starting_window <= longWindowId; prev = temp, temp = temp.next) {
+      for (Block prev = null, temp = first; temp != null && temp.starting_window <= longWindowId;
```
Comment (tweise) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:189]: Don't see the improvement in this case either, line was 130 chars long before.

File: bufferserver/src/test/java/com/datatorrent/bufferserver/server/ServerTest.java [hunk 4]
```diff
@@ -72,7 +72,8 @@ public static void setupServerAndClients() throws Exception
 
     instance = new Server(0, 4096,8);
     address = instance.run(eventloopServer);
-    assert (address instanceof InetSocketAddress);
+    assertTrue(address instanceof InetSocketAddress);
+    assertFalse(address.isUnresolved());
```
Comment (tweise) [inline bufferserver/src/test/java/com/datatorrent/bufferserver/server/ServerTest.java:76]: Why was there code for dealing with unresolved address before?

Comment (vrozov) [inline bufferserver/src/test/java/com/datatorrent/bufferserver/server/ServerTest.java:76]: I think assert for InetSocketAddress (address is declared as InetSocketAddress) and dealing with unresolved address were parts of the original code that was later refactored. Server.run() returns resolved InetSocketAddress.

File: bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java (hunk 3 above)
Comment (vrozov) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:189]: It depends on the width of the monitor. When I don't use external monitor, wrapping at 120 char length improves readability.

File: bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java (hunk 2 above)
Comment (vrozov) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:152]: Not sure what we can do with the string literals that exceed 120 chars.

Comment (davidyan74) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:152]: I agree with Thomas on this. I think in this case, we should break it after the comma and not break the string literal, even if the string literal makes the line over 120 chars.

Comment (vrozov) [inline bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java:152]: I disagree that we should allow exceptions to code style rules. Either 120 line length needs to be enforced or it should not be enforced at all, not just for string literals. Note that some string literals may be very long and span multiple lines.

**Pull request merged**